kinzhal-processor/src/jvmMain/kotlin/com/daugeldauge/kinzhal/processor/KspUtils.kt

```diff
@@ -15,6 +15,7 @@ internal fun KSTypeReference.resolveToUnderlying(): KSType {
         candidate = declaration.type.resolve()
         declaration = candidate.declaration
     }
+    assert(!candidate.isError) { "Unable to resolve type reference: $this" }
```

Comment:
Thank you for the contribution!

Could you please:
1) Add test for this behavior (tests for errors are in https://github.com/daugeldauge/kinzhal/blob/master/kinzhal-processor/src/jvmTest/kotlin/com/daugeldauge/kinzhal/processor/KinzhalSymbolProcessorTest.kt)
2) Replace `assert` with `require` (we don't want to depend on jvm options here)

Reply:
After https://github.com/daugeldauge/kinzhal/commit/0faab4d9af4fc688d2464b18493370098a9dbc06 it'd be better to throw `NonRecoverableProcessorException` with specified node instead of assert/require